src/Transform/ONNX/ConstPropHelper.cpp

```diff
@@ -140,22 +141,18 @@ DenseElementsAttr createDenseElementsAttrFromArray(char *arr, Type outputType) {
   int64_t sizeInBytes = getSizeInBytes(outputType);
   RankedTensorType resType =
       constructRankedTensorType(outputType.cast<ShapedType>());
-  bool isSplat;
-  if (resType.getShape().size() == 0)
-    isSplat = true;
-  else if (llvm::all_of(
-               resType.getShape(), [](int64_t dim) { return dim == 1; }))
-    isSplat = true;
-  else
-    isSplat = false;
+  bool detectedSplat;
+  assert(DenseElementsAttr::isValidRawBuffer(
+             resType, ArrayRef<char>(arr, sizeInBytes), detectedSplat) &&
+         "The raw buffer is invalid for provided type");
   return DenseElementsAttr::getFromRawBuffer(
-      resType, ArrayRef<char>(arr, sizeInBytes), /*isSplat=*/isSplat);
+      resType, ArrayRef<char>(arr, sizeInBytes), detectedSplat);
 }
 
 /// Create a dense ONNXConstantOp from a byte array.
 ONNXConstantOp createDenseONNXConstantOp(PatternRewriter &rewriter,
```

Comment:
Should we add this to the builder? And directly put all the cases in it (meaning `convertDoubleInt64ToExactType`)?
It feels like there is a lot of functions calling one that does some of the work, and then another one that does a bit more. If feel like a `create.onnx.DenseConstant(typeFrom, typeTo, *array)` would be useful abstraction

Reply:
Good suggestion! I will move some functions to the builder. Thanks!